src/sentry/digests/utils.py

```diff
@@ -97,10 +99,20 @@ def get_event_from_groups_in_digest(digest: Digest) -> Iterable[Event]:
     }
 
 
-def build_custom_digest(original_digest: Digest, events: Iterable[Event]) -> Digest:
+def build_custom_digest(
+    original_digest: Digest, events: Iterable[Event], participant: RpcActor
+) -> Digest:
     """Given a digest and a set of events, filter the digest to only records that include the events."""
     user_digest: Digest = {}
     for rule, rule_groups in original_digest.items():
+        skip = False
+        rule_snoozes = RuleSnooze.objects.filter(Q(rule=rule))
+        for snooze in rule_snoozes:
+            if snooze.user_id is None or snooze.user_id == participant.id:
+                skip = True
+                break
+        if skip:
+            continue
```

Comment:
Nit: This is a case for a `for ... else`

``` 
for snooze in rule_snoozes:
    if snooze.user_id is None or snooze.user_id == participant.id:
        break
else:
    continue
```

Doesn't really matter it's just rare that to actually have a use for it. The else runs if we never break.

Reply:
🤔 I tried this out and while it does break the inner loop of the snooze, it just continues to add stuff to the digest that I don't want

Reply:
Oh I think I got it backwards... Yeah, I don't think it'll work. Just got overexcited to use a for/else

Reply:
Lol I'll try to sneak it in somewhere, someday, and when I do, I'll tag you.